pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -78,18 +80,21 @@ func Add(mgr manager.Manager) error {
 type ManifestProvider func() (automationconfig.VersionManifest, error)
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, manifestProvider ManifestProvider) reconcile.Reconciler {
+func newReconciler(mgr manager.Manager, manifestProvider ManifestProvider) *ReplicaSetReconciler {
```

Comment:
We should update this comment? (or maybe get entirely rid of it? it's quite obvious what the function does)

Reply:
Definitely should! I got rid of it.